src/app/core/configurable-enum/basic-autocomplete/basic-autocomplete.component.ts

```diff
@@ -230,48 +239,46 @@ export class BasicAutocompleteComponent<O, V = O>
   }
 
   onFocusIn() {
-    clearTimeout(this.delayedBlur);
-    if (!this.focused) {
-      if (this.multi) {
-        this.autocompleteForm.setValue("");
-      } else {
-        this.showAutocomplete();
-      }
-      this.focus();
+    if (this.multi) {
+      this.autocompleteForm.setValue("");
+    } else {
+      this.showAutocomplete();
     }
+    setTimeout(() => this.inputElement.focus());
```

Comment:
let's comment extensively why we need a timeout here and what are the tradeoffs (as we might face some related bugs again in the future :wink: )